From 32d41ab91e5844ad6df1eea353c61a8fefc0c81e Mon Sep 17 00:00:00 2001
From: Holger Freyther <ich@tamarin.(none)>
Date: Tue, 13 May 2008 18:53:40 +0100
Subject: [PATCH] [pcf50633] Assume that all gta02's have a battery with coulumb counter
     For the gta02 and the bq27000 battery it does not make sense to use the
     ADC to get the current voltage. Under the assumption that all mass
     production gta02's have such batteries it does not make any sense to
     forward this value to APM.

Signed-Off-By: Holger Freyther <zecke@openmoko.org>
---
 drivers/i2c/chips/pcf50633.c |    5 ++---
 1 files changed, 2 insertions(+), 3 deletions(-)

diff --git a/drivers/i2c/chips/pcf50633.c b/drivers/i2c/chips/pcf50633.c
index b354a9c..66ca16f 100644
--- a/drivers/i2c/chips/pcf50633.c
+++ b/drivers/i2c/chips/pcf50633.c
@@ -1797,16 +1797,15 @@ static int pcf50633_detect(struct i2c_adapter *adapter, int address, int kind)
 		backlight_update_status(data->backlight);
 	}
 
-	apm_get_power_status = pcf50633_get_power_status;
 
-#ifdef CONFIG_MACH_NEO1973_GTA02
 	if (machine_is_neo1973_gta02()) {
 		gta01_pm_gps_dev.dev.parent = &new_client->dev;
 		gta01_pm_bt_dev.dev.parent = &new_client->dev;
 		platform_device_register(&gta01_pm_bt_dev);
 		platform_device_register(&gta01_pm_gps_dev);
+	} else {
+		apm_get_power_status = pcf50633_get_power_status;
 	}
-#endif
 
 	return 0;
 exit_rtc:
-- 
1.5.6.5

